sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CollectionQuerySimplification.cs

```diff
@@ -38,6 +38,8 @@ public CollectionQuerySimplification(List<object> coll)
 
             x = Enumerable.Select(coll, element => element as object).Any(element => element != null); //Noncompliant
             x = Enumerable.Any(Enumerable.Select(coll, element => element as object), element => element != null); //Noncompliant
+
+            coll.ToList().AsEnumerable(); // Compliant, we ignore AsEnumerable() as it is somewhat cleaner way to cast to IEnumerable<T> and has no side effects
```

Comment:
Please add more tests with `AsEnumerable` in different locations.

Reply:
It should not be needed - the rule raises when ToList() is called before other Linq method.